backend/uclapi/common/views.py

```diff
@@ -0,0 +1,53 @@
+
+from django.db import connections
+from django.core.cache import cache
+from common.helpers import PrettyJsonResponse
+
+from datetime import datetime, timedelta
+from time import sleep, time
+from uclapi.celery import ping_sample_task
+
+
+def ping_view(request):
+    if not request.GET.get('checks'):
+        return PrettyJsonResponse({
+            "success": True,
+        })
+
+    database_connections = {}
+    for connection in connections:
+        try:
+            connections[connection].cursor()
+        except Exception as e:
+            database_connections[connection] = str(e)
+        else:
+            database_connections[connection] = "Connected"
+
+    cache_connection = False
+    try:
+        cache.set("PING-TEST-CACHE", True, 30)
+        cache_connection = cache.get("PING-TEST-CACHE")
+    except Exception:
+        pass
+
+    celery_connection = False
+    start = time()
+    expires = datetime.now() + timedelta(seconds=10)
+    try:
+        task = ping_sample_task.apply_async(expires=expires)
```

Comment:
This would only test the generic worker, but not test the gencache worker

Reply:
Sorry that took so long, fixed now